### PR TITLE
[#185] Add golden tests

### LINKED ISSUE #185: Add golden tests for `summoner`
I propose the following idea for golden tests:

1. Create projects with all options specified.
2. Copy content of this project somewhere in `test/golden/`
3. Implement automatic tests that will generate project with same settings and will compare content of every file.

### PR DESCRIPTION
**Pull request by vrom911**
Resolves #185

## Checklist:

- [x] I've updated the [CHANGELOG](https://github.com/kowainik/summoner/blob/master/CHANGELOG.md) with the short description of my latest changes.
- [x] All new and existing tests pass.
- [x] I keep the code style used in the files I've changed (see [style-guide](https://github.com/kowainik/org/blob/master/style-guide.md#haskell-style-guide) for more details).
- [x] I've used the [`stylish-haskell` file](https://github.com/kowainik/summoner/blob/master/.stylish-haskell.yaml).
- [ ] My change requires the documentation updates.
  - [ ] I've updated the documentation accordingly.
- [x] I've added the `[ci skip]` text to the docs-only related commit's name.
- [ ] I changed code related to the generated project.
  - [ ] I created a new project using `summoner` and I checked that this specific feature is working as expected.

### CODE REVIEW

File: test/Test/QuestionSpec.hs [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
-import Test.Tasty.HUnit ((@?=))
+import Summoner.Question (YesNoPrompt (..), mkDefaultYesNoPrompt)
 
-import Summoner.Question
-
-unit_mkDefaultYesNoPrompt :: IO ()
-unit_mkDefaultYesNoPrompt =
-  yesNoPrompt (mkDefaultYesNoPrompt "Cabal") @?= "Add Cabal?"
+yesNoPromptSpec :: Spec
+yesNoPromptSpec = describe "mkDefaultYesNoPrompt works correctly" $
```
Comment (willbasky) [inline test/Test/QuestionSpec.hs]: I wonder does it possible to make that names better. :thinking:

File: test/Spec.hs [hunk 2]
```diff
@@ -1 +1,26 @@
...
+import Test.DecisionSpec (decisionMonoidMempty, decisionSemigroupAssoc)
+import Test.Golden (goldenSpec)
+import Test.QuestionSpec (yesNoPromptSpec)
+import Test.TomlSpec (tomlProp)
+
+
+main :: IO ()
+main = do
+    hspec yesNoPromptSpec
+    hspec goldenSpec
```
Comment (ChShersh) [inline test/Spec.hs]: You can write just:
```haskell
hspec $ do
    yesNoPromptSpec
    goldenSpec
```

Though it's one more extra line... Alternative you create list of specs  with something like:

```haskell
hspec $ sequence_ [yesNoPromptSpec, goldenSpec]
```

File: test/Test/DecisionSpec.hs [hunk 3]
```diff
@@ -1,22 +1,27 @@
...
+decisionSemigroupAssoc = property $ do
     x <- forAll genDecision
     y <- forAll genDecision
     z <- forAll genDecision
     (x <> y) <> z === x <> (y <> z)
 
-test_decisionMonoidMempty :: [TestTree]
-test_decisionMonoidMempty = pure $ testProperty "Decision Monoid:Mempty" $ property $ do
+decisionMonoidMempty :: Property
+decisionMonoidMempty = property $ do
```
Comment (ChShersh) [inline test/Test/DecisionSpec.hs]: Thank you for the renaming! I don't like `tasty` naming scheme so much..

File: test/Test/QuestionSpec.hs (hunk 1 above)
Comment (vrom911) [reply test/Test/QuestionSpec.hs]: Do you have any suggestions?

Comment (willbasky) [reply test/Test/QuestionSpec.hs]: My suggestions are not golden yet. So let it be like they are. Now they are our the best.

**Pull request merged**